### PR TITLE
converted test_sync to using pytest

### LINKED ISSUE #9: Convert test_sync.py to pytest
https://github.com/hofbi/dev-sync/blob/master/tests/test_sync.py currently uses [unittest](https://docs.python.org/3/library/unittest.html) but we want to convert it to using [pytest](https://docs.pytest.org/en/8.0.x/).

- [ ]  Setup the project as described in https://github.com/hofbi/dev-sync?tab=readme-ov-file#development.
- [ ]  Convert https://github.com/hofbi/dev-sync/blob/master/tests/test_sync.py to pytest. https://github.com/hofbi/dev-sync/blob/master/tests/test_args.py can be used as an example for something that is already converted to pytest. Also take a look https://pytest-subprocess.readthedocs.io/en/latest/ for replacing the subprocess mocks with the fakeprocess

### PR DESCRIPTION
**Pull request by dyxu10**
Closes https://github.com/hofbi/dev-sync/issues/9

### CODE REVIEW

File: tests/test_sync.py [hunk 1]
```diff
@@ -1,64 +1,66 @@
...
+    assert "-n" not in RSync.get_options(False, [])
+
+
+def test_get_options_no_excludes() -> None:
+    assert "--exclude" not in RSync.get_options(False, [])
+
+
+def test_get_options_2_excludes() -> None:
+    assert '--exclude="/tmp/foo"' in RSync.get_options(False, [Path("/tmp/foo"), Path("/tmp/bar")])
+    assert '--exclude="/tmp/bar"' in RSync.get_options(False, [Path("/tmp/foo"), Path("/tmp/bar")])
```
Comment (hofbi) [inline tests/test_sync.py:22]:
```suggestion
    options = RSync.get_options(False, [Path("/tmp/foo"), Path("/tmp/bar")])
    assert '--exclude="/tmp/foo"' in options
    assert '--exclude="/tmp/bar"' in options
```

File: .DS_Store
Comment (hofbi) [inline .DS_Store:1]: Please remove this file and add it to the `.gitignore`

**Review by hofbi — COMMENTED** [review]
Please also check why the CI failed

**Comment by dyxu10** [comment]
replace fake process with monkey patch, adjusted gitignore to ignore OS-generated files

**Comment by hofbi** [comment]
Could you try to use the fakeprocess. Please check the error message in CI, it should tell you what went wrong

**Comment by dyxu10** [comment]
i added pytest_subprocess to requirements because the fake_process fixture was not found by ci

File: tests/test_sync.py [hunk 2]
```diff
@@ -1,64 +1,49 @@
...
-
-    def test_get_options_2_excludes(self):
-        self.assertIn(
-            '--exclude="/tmp/foo"',
-            RSync.get_options(False, [Path("/tmp/foo"), Path("/tmp/bar")]),
-        )
-        self.assertIn(
-            '--exclude="/tmp/bar"',
-            RSync.get_options(False, [Path("/tmp/foo"), Path("/tmp/bar")]),
-        )
```
Comment (hofbi) [inline tests/test_sync.py:32]: What happended to these tests?

**Comment by dyxu10** [comment]
I added the tests back. sorry for uploading the wrong file!

File: tests/test_sync.py (hunk 2 above)
Comment (dyxu10) [reply tests/test_sync.py:32]: sorry, I uploaded the wrong file

File: .DS_Store
Comment (hofbi) [reply .DS_Store:1]: This file still has to be removed

File: tests/test_sync.py (hunk 1 above)
Comment (hofbi) [reply tests/test_sync.py:22]: This one is still/again open

File: tests/test_sync.py [hunk 3]
```diff
@@ -1,64 +1,66 @@
...
+    monkeypatch.setattr(subprocess, "check_call", mock_check_call)
+
+    backup_folder = BackupFolder(Path("/foo"), "blub")
+    backup_folder.get_relative_repo_paths = list
+
+    rsync = RSync(Path("/foo"), [backup_folder, backup_folder, backup_folder])
+    rsync.sync(Target("/tmp"), False)
+
+    count = 3
+    assert mock_check_call.call_count == count
```
Comment (hofbi) [inline tests/test_sync.py:55]: Can we solve this with the fake process as well?

**Comment by hofbi** [comment]
One new comment, and 2 older ones that still need to be addressed

File: tests/test_sync.py (hunk 3 above)
Comment (dyxu10) [reply tests/test_sync.py]: I adjusted test_sync_three_backup_folders to use fake_process as well. It was originally giving me an error that said the command was not registered when I did fake_process.register(command). It took me some time to figure out that I needed to specify the occurrence to be three.

**Pull request merged**